src/AWSCredentials.jl

```diff
@@ -410,10 +410,25 @@ function dot_aws_config(profile=nothing)
     if isfile(config_file)
         ini = read(Inifile(), config_file)
         p = @something profile _aws_get_profile()
-        access_key, secret_key, token = _aws_get_credential_details(p, ini)
 
-        if access_key !== nothing
-            return AWSCredentials(access_key, secret_key, token)
+        # get all the fields for that profile
+        settings = _aws_profile_config(ini,p)
+
+        if isempty(settings)
+            return nothing
+        end
```

Comment:
One of the coding styles that we use in this package to make things a bit smaller. There's another instance down in the credential utilities files as well.

```suggestion
        settings = _aws_profile_config(ini, p)
        isempty(settings) && return nothing
```